src/overlays/actors/ovl_En_Sw/z_en_sw.c

```diff
@@ -35,95 +33,1003 @@ const ActorInit En_Sw_InitVars = {
     (ActorFunc)EnSw_Draw,
 };
 
-static ColliderJntSphElementInit D_80B0F040[1] = {
+static ColliderJntSphElementInit sJntSphItemsInit[1] = {
     {
-        {
-            ELEMTYPE_UNK0,
-            { 0xFFCFFFFF, 0x00, 0x08 },
-            { 0xFFC3FFFE, 0x00, 0x00 },
-            TOUCH_ON | TOUCH_SFX_NORMAL,
-            BUMP_ON | BUMP_HOOKABLE,
-            OCELEM_ON,
-        },
+        { ELEMTYPE_UNK0, { 0xFFCFFFFF, 0x00, 0x08 }, { 0xFFC3FFFE, 0x00, 0x00 }, 0x01, 0x05, 0x01 },
         { 2, { { 0, -300, 0 }, 21 }, 100 },
     },
 };
 
-static ColliderJntSphInit D_80B0F064 = {
-    {
-        COLTYPE_HIT6,
-        AT_ON | AT_TYPE_ENEMY,
-        AC_ON | AC_TYPE_PLAYER,
-        OC1_ON | OC1_TYPE_ALL,
-        OC2_TYPE_1,
-        COLSHAPE_JNTSPH,
-    },
+static ColliderJntSphInit sJntSphInit = {
+    { COLTYPE_HIT6, 0x11, 0x09, 0x39, 0x10, COLSHAPE_JNTSPH },
     1,
-    D_80B0F040,
+    sJntSphItemsInit,
+};
+
+static CollisionCheckInfoInit2 D_80B0F074 = { 0x01, 0x0002, 0x0019, 0x0019, 0xFF };
+
+static struct_80034EC0_Entry D_80B0F080[] = {
+    { 0x06000304, 1.0f, 0.0f, -1.0f, 0x01, 0.0f },
+    { 0x06000304, 1.0f, 0.0f, -1.0f, 0x01, -8.0f },
+    { 0x060055A8, 1.0f, 0.0f, -1.0f, 0x01, -8.0f },
+    { 0x06005B98, 1.0f, 0.0f, -1.0f, 0x01, -8.0f },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0BDB0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Sw/func_80B0BE20.s")
+char D_80B0F630[0x80]; // unused.
+
+extern SkeletonHeader D_06005298;
+extern Gfx D_06004788[];
+extern Gfx D_060046F0[];
+extern Gfx D_06004658[];
+extern Gfx D_060045C0[];
+extern Gfx D_06004820[];
+extern Gfx D_060048B8[];
+extern Gfx D_06004950[];
+extern Gfx D_060049E8[];
+extern Gfx D_06003FB0[];
+extern Gfx D_060043D8[];
+extern AnimationHeader D_06000304;
+
+void EnSw_CrossProduct(Vec3f* a, Vec3f* b, Vec3f* dst) {
+    dst->x = (a->y * b->z) - (a->z * b->y);
+    dst->y = (a->z * b->x) - (a->x * b->z);
+    dst->z = (a->x * b->y) - (a->y * b->x);
+}
+
+s32 func_80B0BE20(EnSw* this, CollisionPoly* poly) {
+    Vec3f sp44;
+    Vec3f sp38;
+    f32 sp34;
+    f32 temp_f0;
+    s32 pad;
+
+    this->actor.floorPoly = poly;
+    sp44.x = poly->normal.x * (1.0f / 32767.0f);
```

Comment:
```suggestion
    sp44.x = poly->normal.x * COLPOLY_NORM_FRAC;
```

Reply:
Actually, do not use COLPOLY_NORM_FRAC. Use the COLPOLY_GET_NORMAL macro instead: 
```suggestion
    sp44.x = COLPOLY_GET_NORMAL(poly->normal.x);
```